File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int l, r;
        l = 0;

        int counter = 0;
        while (l < nums.size()) {
            r = l;
            int running_sum = 0;
            while (r < nums.size()) {
                running_sum += nums[r++];
                if (running_sum == k) counter++;
            }
            l++;
        }
        return counter;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0: return 0
        res = []
        nums = sorted(nums)
        for i in range(len(nums)):
            unique = set()
            unique.add(nums[i])
            seq = [nums[i]]
            for j in range(i + 1, len(nums)):
                if nums[j] in unique: continue
                unique.add(nums[j])
                if nums[j] != seq[-1] + 1: break
                seq.append(nums[j])
            res.append(len(seq))
        return max(res)
